src/public/audits-table.vue

```diff
@@ -0,0 +1,261 @@
+                    <span class="text-greyish">{{ scope.row.created_at | fixDateTime }}</span>
+                </template>
+            </el-table-column>
+
+            <template slot="empty">
+                <span v-if="!loading">No Data</span>
+                <span v-if="loading"></span>
+            </template>
+        </el-table>
+
+        <div class="row mt-3">
+            <div class="col-12">
+                <el-button-group class="pull-right">
+                    <el-button size="small"
+                               icon="el-icon-arrow-left"
+                               v-if="!pagination.prev_page_url || pagination_loading || loading"
+                               disabled
+                               plain>
+                        Previous Page
+                    </el-button>
+                    <el-button type="success"
+                               size="small"
+                               icon="el-icon-arrow-left"
+                               v-else
+                               @click="previousPage">
+                        Previous Page
+                    </el-button>
+
+                    <el-button size="small"
+                               v-if="!pagination.next_page_url || pagination_loading || loading"
+                               disabled
+                               plain>
+                        Next Page
+                        <i class="el-icon-arrow-right"></i>
+                    </el-button>
+                    <el-button type="success"
+                               size="small"
+                               v-else
+                               @click="nextPage">
+                        Next Page
+                        <i class="el-icon-arrow-right"></i>
+                    </el-button>
+                </el-button-group>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+export default {
+    props: {
+        model_alias: {
+            type: String,
+            required: true
+        },
+        model_id: {
+            required: true
+        },
+        action_label_resolver: {
+            type: Function,
+            required: false
+        }
+    },
+
+    data() {
+        return {
+            loading: false,
+            items: [],
+            pagination: {},
+            pagination_loading: false,
+            table_height: 769,
+            extra_height: 0,
+            source: null,
+            filter: {}
+        }
+    },
+
+    created() {
+        this.CancelToken = axios.CancelToken
+        this.source = this.CancelToken.source()
+
+        this.getAudits()
+    },
+
+    methods: {
+        actionLabels(audit) {
+            if (this.action_label_resolver) {
+                return this.action_label_resolver(audit);
+            }
+
+            return [audit.event_type + ' - ' + audit.label];
+        },
+
+        getAudits(url = null) {
+            url = url || '/api/audits/line/' + this.model_id
```

Comment:
Why is there a default here? It makes no sense. Don't make it nullable; UI should request this if we want to use it as a package.

Reply:
The package defines its own route to load the audits, that already handles most filtering and pagination. Shall the parent application choose to use its own endpoint, it can. It just offers a generic solution while leaving it open to use a different solution.

I don't see an issue with that. Can you explain why it doesn't make sense?

Reply:
I prefer the package not to have a default URL and rely on a given URL for loading the audits.

Reply:
We need to pass a URL when we run `getAudits()` in the app when we use the package.

Reply:
No, we don't. The possibility is there for flexibility, but the package provides the table, trait, and endpoint, and handles it automatically without any required customization. In particular, no custom url/endpoint is needed.